### ISSUE TITLE
Add More Graphs

### ISSUE THREAD

**Issue by Nick3C** [batterseapower/pinyin-toolkit#118]
Current plans for:
1) Trad/Simp Characters Known - cards using the non-preference simp/trad in the expression (probably need a candidate-list for this) to show the number of non-preference simp/trad known (i.e. graph of traditional characters known)
2) Mature Characters (not quite sure what conditions would be best, maybe interval over 21 days and initiated by 2 or 3 card models)
3) Completely known Characters- i.e. look for facts with all cards possible are > 21 days (not all possible models should be the total -1 if MW empty, -1 more if simp/trad empty] - this would show words that you can actually read/write properly, not just know for one card template
4) Deck Constitution - Facts[not chinese specific] - divded into Completly known, zero known, and partially initalised (some card templates).
5) Deck Constitution - Cards [not chinese specific], all cards by new, young, mature

**Comment by Nick3C** [comment]
Possibly add the traditional graph for 0.06 (which should be the easiest to add) as it goes with the new traditional feature.

Others can wait

**Comment by Nick3C** [comment]
Max, why does the plugin object to me adding:
    (u'The 214 Bushou Radicals', u'一丨丶丿乙亅二亠人儿入八冂冖冫几凵刀力勹匕匚匸十卜卩厂厶又口囗土士夊夊夕大女子宀寸小尢尸屮山巛工己巾乡广廴廾弋弓彐彡彳心戈戶手支攴文斗斤方无日曰月木欠止歹殳毋比毛氏气水火爪父爻爿片牙犬玄玉瓜瓦甘生用田疋疒癶白皮目矛矢石示禸禾穴立竹米糸缶网羊羽老而耒耳聿肉臣自至臼舌舛舟艮色艸虍虫血行衣襾見角言谷豆豕豸貝赤走足身車辛辰辵邑酉釆里金長門阜隶隹雨靑非面革韦韭音頁凬飛食首香馬骨高髟鬥鬯鬲鬼魚鳥鹵鹿麦麻黃黍黑黹黽鼎鼓鼠鼻齊'),

to 'hanziByGrades' in statistics.py?

I wanted to add that and then remove the characters in the radicals from the other sets.

**Comment by Nick3C** [comment]
never mind, I got it (had to edit statsandgraph.py as well)

**Comment by batterseapower** [comment]
I see you added radicals. However, you haven't removed the characters that are present in the other sets from the radical list, which means the statistics are going to be broken.

BTW: if we really must include radicals, can you please remove things from the RADICALS list, not the HSK ones (and rename the radicals list "Non-HSK Radicals"?). Personally I'm more interested in knowing my progress towards HSK than knowing what percentage of useless radicals I know.

Please can you fix this or comment out the radicals stuff?

**Comment by Nick3C** [comment]
I was originally planning to remove the radicals from the hsk list but then opted not to do so because the code seems to just match the first instance and return it. As the radicals are first it grabs them without ever looking at the hsk levels. How are the statistics broken?

The suggestion of removing things from the radicals list is not really feasible. It would render the statistic totally useless. Consider the 一 radical (which is in the hsk list). It doesn't make sense for it to be listed in hsk basic and not listed in the racial set as a user would expect it to be a simple radical. Moreso the 214 radicals only really have relevance as a set. It is also a small set, cutting chunks out of it will also diminish its usefulness greatly.

I think we may have an ideological difference here. I see the graph as a motivational tool for learning 'more' characters. You seem to be suggesting that you consider the aim to learn the hsk characters themselves. Adding the radicals assists the former objective but hinders the latter.

The issue for me is that learning the 214 radicals is an achievement for most learners (which is much easier than learning the first level of hsk characters). Motivationally it is better for them to learn the radicals first (in the same way it has been better for you to learn the rthz radicals first). Something you are encouraged to do as a relatively easy first-objective to give you a couple of hundred characters. Learning HSK characters, by contrast, isn't an end in itself. First of all they are not necessarily the characters you should learn first; it is not a frequency list so they are not the most useful ones. You still can't read hsk texts because you need to be learning hsk words not characters (and there are around 10 times the number of words as there are character [a guess]).

I am more than willing to compromise on this issue. If you insist we can just remove the radicals. I would rather this than cut out radicals from the 214 (which I think is a bad idea).

However, perhaps we can find a better way to keep us both happy and still motivate learners. How about we try something new, changing the current 1 graph structure to have 3 graphs:
1) cumulative characters in deck (using divisions of 500s from the hanzi stats plugin) showing ALL characters to replace the current graph
2) a radicals graph showing only things we deem to be simple radcials [we could start with the 214 and add other radical sets over time, obiously the RtHZ set is a prime candidate)
3) graph showing HSK characters to track this progress (showing only basic, elementary, intermediate, and advanced) NOT showing non-hsk. This would reduce clutter and allow users to 'complete' their objective of learning hsk characters.

This would mean that the radical and hsk graphs were used as specific learning goals a user could pick and then work towards. The 500s graph would serve the purpose of the current graph (overall progress) but in a more neutral way of showing which characters are known..

What do you think?

**Comment by Nick3C** [comment]
In fact, if we take this route it might be better to take it even further and separate each HSK level to its own graph and just have the 500-divisions in the main graph. This would solve our zoom type issue too and reinforce the idea of picking a goal and working to achieve it.

The other thing I should have mentioned is that the nearly 803 characters in the basic hsk level is really quite large and could do with being broken down. That's why I thought the 214 radicals was a good way to do this. I guess double coverage is probably about 50 to 100.

**Comment by Nick3C** [comment]
Just to add one more point. Despite knowing 2200 characters my hsk stats read (99.1, 79.82, 47.82)). On the other hand my 500 statistics show much better figueres (100, 99.6, 77,56.6,35.5,21.2,10.4 + 188). 

Even though I have some doubts about the frequency list used by hanzi stats I am really sure the list is more useful than the hsk list in terms of frequency and usefullness. Some of the basic hsk characters are really not either simpl or common).

If you want I am sure I can dig up some criticisms of the hsk work lists, there has been past discussion of this in relation to mdbg.

**Comment by Nick3C** [comment]
As luck would have it a recent forum post has drawn my attenton to this thread:
http://forum.koohii.com/viewtopic.php?id=3513
and http://residence.educities.edu.tw/wei3128/currinstruc/wordclause/generwordgrd9.htm

It provides an alternative list of characters (much larger and more comprehensive than HSK) that is used in Taiwan. 
Level 1: 198
Level 2: 166
Level 3: 267
Level 4: 351
Level 5: 421
Level 6: 548
Level 7: 616
Level 8: 956
Level 9: 2042

I also like the idea of smaller intervals between the levels.

How about we implement this as an additional stats package to the HSK? There is obviously a question as to how we would do this in terms of trad/simp. I propose that we actually need to run the stats in terms of the trad/simp preference (thus defining the character sets in the alternative) and we then just use the non-preference set graph in 500 character intervals.

Any thoughts?

**Comment by Nick3C** [comment]
I've changed it back to the previous behaviour but added several new catagories of tag with a view to future graphing:
1) radicals(214 and rthz)
2) hanzi by 500s [trad and simp]
3) taiwan standard

2 is from the hanzi stats plugin. We can go with this for now and consider revising the actual character choice at a later point.

**Comment by batterseapower** [comment]
Yes, having separate graphs seems a bit more reasonable to me. We will never find a way to show info. about all these classifications (HSK, Taiwan, frequency, radicals) on the /same/ graph in a way we can agree about :-)
